hazelcast-jet-core/src/test/java/com/hazelcast/jet/pipeline/test/TestSourcesTest.java

```diff
@@ -53,16 +55,16 @@ public void test_items() {
     public void test_longStream() throws Throwable {
         int itemsPerSecond = 250;
         int timeoutSeconds = 10;
-        int numWindowResultsToWaitFor = 3;
+        int numWindowResultsToWaitFor = 5;
 
         p.readFrom(TestSources.longStream(itemsPerSecond, 0))
          .withNativeTimestamps(0)
          .window(tumbling(SECONDS.toMillis(1)))
          .aggregate(counting())
-         .apply(assertCollectedEventually(timeoutSeconds, windowResults -> {
-             assertTrue("Didn't receive at least three results", windowResults.size() >= numWindowResultsToWaitFor);
-             assertTrue("Invalid items per second",
-                     windowResults.stream().skip(1).allMatch(wr -> wr.result() == itemsPerSecond));
+         .apply(assertCollectedEventually(timeoutSeconds, results -> {
+             assertTrue("Didn't receive at least three results", results.size() >= numWindowResultsToWaitFor);
+             assertTrue("Invalid items per second: " + results.stream().map(WindowResult::result).collect(toList()),
+                     results.stream().skip(2).allMatch(wr -> wr.result() == itemsPerSecond));
```

Comment:
Why skipping two instead of 1 helps?

Reply:
I don't exactly know, but that's how it was failing. I'm assuming that in the problematic high system load scenarios it just takes a bit longer for the source instances to get up to speed. I've never seen the 3rd or further windows not have the perfect number of items in them.

Reply:
"Seems to work"
:-)

Reply:
How is such a test supposed to be deterministic in the first place? How can a source be expected to produce a certain rate of events, regardless of how overloaded the machine running it is? So yes, the non-perfect test seems to be working a bit better like this.